icicle-source/src/Icicle/Source/Transform/ReifyPossibility.hs

```diff
@@ -296,15 +296,7 @@ makeApps a fun (arg:rest) doWrap
 
  -- If argument is a definitely, just apply it as usual
  | otherwise
- =  makeApps a (App annotWrapPrim fun arg) rest doWrap
-
- where
-  annotWrapPrim
-   | Just (p, _, _) <- takePrimApps fun
-   , primReturnsPossibly p
-   = wrapAnnotReally a
-   | otherwise
-   = a
+ =  makeApps a (App a fun arg) rest doWrap
```

Comment:
This is a rejigging of https://github.com/ambiata/icicle/commit/130a02bd2d211743960724e72afd8d2da6824051
We were wrapping the application annotations with `SumT ErrorT`, but not the primitive itself